samples/CosmosDbIntegtation/ReadThroughSamples/StreamSample.cs

```diff
@@ -0,0 +1,79 @@
+﻿using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Linq;
+using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples
+{
+    internal class StreamSample
+    {
+        // Redis connection string and stream names stored in local.settings.json
+        public const string redisConnectionSetting = "redisConnectionString";
+        private static readonly Lazy<IDatabaseAsync> redisDB = new Lazy<IDatabaseAsync>(() =>
+           ConnectionMultiplexer.Connect(Environment.GetEnvironmentVariable(redisConnectionSetting)).GetDatabase());
+        private const string streamNameSingleDocument = "streamTest_WriteBehind";
+
+        // CosmosDB connection string, client, database name and container name stored in local.settings.json
+        public const string cosmosDbConnectionSetting = "cosmosDbConnectionString";
+        public const string databaseSetting = "%cosmosDbDatabaseId%";
+        public const string containerSettingSingleDocument = "%cosmosDbContainerIdSingleDocument%";
+
+        /// <summary>
+        /// Read Through: If the stream does not exits, refresh it with the values saved in CosmosDB (only available if entries were written in a single document)
+        /// </summary>
+        /// <param name="entry"> The message which has gone through the stream. Includes message id alongside the key/value pairs </param>
+        /// <param name="items"> Container for where the CosmosDB items are stored </param>
+        /// <param name="logger"> ILogger used to write key information </param>
+        /// <returns></returns>
+        [FunctionName(nameof(ReadThroughForStreamSingleDocumentAsync))]

```

Comment:
For my own learning, I'm not sure what this line does, could you explain it to me :)